packages/harpguru-core/src/components/menu-open-button/hooks/use-tap-animation/use-tap-animation.ts

```diff
@@ -0,0 +1,44 @@
+import { useTimingTransition } from 'react-native-redash'
+import { Node, Easing, interpolate } from 'react-native-reanimated'
+import { State } from 'react-native-gesture-handler'
+import type { TapGestureHandlerStateChangeEvent } from 'react-native-gesture-handler'
+import React from 'react'
+
+import { usePrevious } from '../../../../hooks'
+
+type TapEventHandler = (arg0: TapGestureHandlerStateChangeEvent) => void
+
+export const useTapAnimation = (
+  openCloseMenu: () => void
+): [Node<number>, TapEventHandler] => {
+  const [isTapped, setIsTapped] = React.useState(false)
+  const changedTap = isTapped !== usePrevious(isTapped, isTapped)
```

Comment:
If this is necessary here then presumably it's necessary in the other hooks too. The reason this is needed is to prevent an update to the `openCloseMenu` dependency of the `useEffect` hook producing a toggle action.

Possibly it's not needed by all the other  tap handler logics, but it's worth considering.

Reply:
No it's not necessary. Removed